unicodetools/src/main/java/org/unicode/text/UCD/TestUnicodeInvariants.java

```diff
@@ -571,6 +593,87 @@ static UnicodeProperty of(
                     propOrFilter.filter = parseUnicodeSet(line, pp);
                     propOrFilter.type = FilterOrProp.Type.filter;
                     result.propOrFilters.add(propOrFilter);
+                } else if (line.charAt(pp.getIndex()) == '(') {
+                    final FilterOrProp propOrFilter = new FilterOrProp();
+                    final var matcher =
+                            Pattern.compile("(\\( *([^ )]+)(?: +([^)]+))? *\\)).*")
+                                    .matcher(line.substring(pp.getIndex()));
+                    if (matcher.matches()) {
```

Comment:
Minor, but I find the code more readable if the failure condition is first, rather than disconnected by being at the end of a bunch of code.

if (!matcher.matches()) {
  // handle error
} else {
 //many lines of code
}

Reply:
But your choice